File: Golang/Medium/3217_Delete_Nodes_From_Linked_List_Present_in_Array/solution.go
package deletenodesfromlinkedlistpresentinarray

// Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func modifiedList(nums []int, head *ListNode) *ListNode {
	numMap := make(map[int]int)

	for _, num := range nums {
		numMap[num] += 1
	}

	dummyHead := ListNode{0, nil}
	dummyHead.Next = head
	prev := &dummyHead
	current := dummyHead.Next

	for current != nil {
		if _, exist := numMap[current.Val]; exist {
			prev.Next = current.Next
			current = current.Next
			continue
		}

		prev = current
		current = current.Next
	}
	return dummyHead.Next
}

File: Python/Medium/3217_Delete_Nodes_From_Linked_List_Present_in_Array/solution.py
# Definition for singly-linked list.
from typing import List, Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        numSet = set()
        for i in range(len(nums)):
            numSet.add(nums[i])

        dummyHead = ListNode(0)
        dummyHead.next = head
        prev = dummyHead
        current = dummyHead.next

        while current != None:
            if current.val in numSet:
                prev.next = current.next
                current = current.next
                continue

            prev = current
            current = current.next
        
        return dummyHead.next